src/rewritable_units/element.rs

```diff
@@ -127,6 +127,12 @@ impl<'r, 't> Element<'r, 't> {
         Ok(())
     }
 
+    /// Whether the starting tag of this element is explicitly self-closing, e.g. `<foo />`.
+    #[inline]
+    pub fn is_start_tag_self_closing(&self) -> bool {
```

Comment:
This could just be called `is_self_closing` since it is a property of the element, not just the tag.